### PR TITLE
Android

### PR DESCRIPTION
**Pull request by jonashendrickx**

### CODE REVIEW

**Comment by abergs** [comment]
Since it contains TODO's I'm guessing this is not yet ready for review, marking as draft.

File: src/guide/frontend/android.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+```
+
+### Configuration (Android application)
+
+```kotlin
+data class PasswordlessOptions(
+   # Your public API key
+   val apiKey: String,
+
+   # Identifier for your server, for example 'example.com' if your backend is hosted at https://example.com.
```
Comment (jrmccannon) [inline src/guide/frontend/android.md:60]:
```suggestion
   // Identifier for your server, for example 'example.com' if your backend is hosted at https://example.com.
```

File: src/guide/frontend/android.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+   val rpId: String,
+
+   # This is where your Facet ID goes
```
Comment (jrmccannon) [inline src/guide/frontend/android.md:63]:
```suggestion
   // This is where your Facet ID goes
```

File: src/guide/frontend/android.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+   val origin: String,
+
+   # Where your backend is hosted
```
Comment (jrmccannon) [inline src/guide/frontend/android.md:66]:
```suggestion
   // Where your backend is hosted
```

File: src/guide/frontend/android.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+   val backendUrl:String,
+
+   # Passwordless.dev server, change for self-hosting
```
Comment (jrmccannon) [inline src/guide/frontend/android.md:69]:
```suggestion
   // Passwordless.dev server, change for self-hosting
```

File: src/guide/frontend/android.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+        }
+    } else {
+        showException(exception)
+    }
+}
+```
+
+## Sample
+
+You can find the same application [here](https://www.github.com/bitwarden/passwordless-android).
```
Comment (jrmccannon) [inline src/guide/frontend/android.md:296]: Maybe have this point to the sample app? 
https://github.com/bitwarden/passwordless-android/tree/main/app

And provide a README on how to get it up and running locally

**Review by jrmccannon — DISMISSED** [review]

File: src/guide/frontend/android.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+
+# Android Client SDK
+
+The Passwordless.dev Android client SDK gives users the ability to leverage their device’s built-in fingerprint sensor and/or FIDO security keys for secure passwordless access to websites and native applications that support the FIDO2 protocols
+
+## Creating your first Android application
+
+### Requirements
+
+- Android 7.0 (API level 24) or higher
```
Comment (abergs) [inline src/guide/frontend/android.md:13]: I don't think this version of Android supports Passkeys, I think device bound passkeys were added on Android 9 and the CredMan "modern passkeys were added in Android 14.

https://developers.google.com/identity/passkeys/supported-environments

File: src/.vuepress/config.js
Comment (abergs) [inline src/.vuepress/config.js:1]: question: Why is this commited? We have a config.ts - this seems to be a misstake?

**Review by abergs — CHANGES_REQUESTED** [review]
Nice write-up - lots of good knowledge collected here.

I do think we can iterate on it to make it read similar to the existing guides:

* https://docs.passwordless.dev/guide/frontend/javascript.html
* https://docs.passwordless.dev/guide/frontend/aspnet.html

Suggestion: I think we could add a little bit of "hand holding", and perhaps move the technicalities of facetid and assetlinking further down, first showing installation and usage of the sdk?

**Comment by abergs** [comment]
@fschillingeriv Hey Fred, I'm wondering if you could help us edit this technical documentation for our Android SDK?

While the page current has great detailed knowledge, I feel like we should adapt the way we present it. Not all of it is directly related to Passwordless.dev and is more "creating an android app".

Perhaps @mpbw2 or @fedemkr could also shime in and just point out what we could put in a sub-page and just link to? https://android.passwordless-docs.pages.dev/

File: src/guide/frontend/android.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+    }
+}
+```
+
+## Logging in
+
+1. **UI**: Optionally, take the alias as input from the user if it's a non-discoverable credential.
+2. **Call `PasswordlessClient.login()`**: Initiate the login process with the (optional) alias and response callback.
+3. **Call `POST /signin/verify` implemented in `your backend`**: Your backend should call the Passwordless.dev API to generate a login token.
+   - See (API documentation)[https://docs.passwordless.dev/guide/api.html#signin-verify]
```
Comment (abergs) [inline src/guide/frontend/android.md:290]: Broken link syntax, change to []()

File: src/guide/frontend/android.md [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+## Permissions
+
+In your `AndroidManifest.xml`, you will need to add the following permissions:
+
+```xml
+<uses-permission android:name="android.permission.INTERNET" />
+```
+
+## Configuration (Android application)
+
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:52]: Does it make sense to preface this with information about where to add this data class (e.g. like you do when you say "In your application's AndroidManifest.xml")?

File: src/guide/frontend/android.md [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+```groovy
+implementation 'com.bitwarden:passwordless-android:1.0.1'
+```
+
+:::
+
+## Permissions
+
+In your `AndroidManifest.xml`, you will need to add the following permissions:
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:45]: Small editorial comment; wherever you write ", you will need to add..." can be shortened to ", add..." for brevity's sake.

File: src/guide/frontend/android.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+### .well-known/assetlinks.json
+
+In your application's `AndroidManifest.xml`, you will then need to add the tag below under `manifest::application`:
+
+```xml
+<meta-data
+  android:name="asset_statements"
+  android:resource="@xml/assetlinks" />
+```
+
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:81]: Is it "our" assetlinks.json file or "their" assetlinks.json file? Asking b/c in the example it looks like it's hosted under their domain.

File: src/guide/frontend/android.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+<resources>
+    <string name="assetlinks">https://yourexample.com/.well-known/assetlinks.json</string>
+</resources>
+```
+
+### Facet ID
+
+The `Facet ID` will be used at a later point in this guide to use as the `origin`.
+
+To obtain the Facet ID continue the steps below, the facet id typically looks like:
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:95]: Would suggest capitalizing "ID" to match the others.

File: src/guide/frontend/android.md [hunk 12]
```diff
@@ -0,0 +1,340 @@
...
+3. Now append the result to `android:apk-key-hash:` to get the Facet ID:
+   `android:apk-key-hash:POIplOLeHuvl-XAQckH0DwY4Yb1ydnnKcmhn-jibZbk`
+
+## Configuration (Your back-end)
+
+### Obtaining the SHA-256 Certificate Fingerprints
+
+To configure your backend, you'll need to host a `.well-known/assetlinks.json` file at the root of your domain. This file contains a list of SHA-256 certificate fingerprints that are allowed to authenticate with your backend.
+
+This command will print detailed information about the keystore entry with the specified alias, including information about the certificate, its validity, and other details. It's commonly used in Android development to verify the properties of the debug keystore and the associated key used for signing applications during development.
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:148]: Would suggest changing "This command" to "The following command" to draw attention to the fact that you're describing what's in the tabs below.

File: src/guide/frontend/android.md [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+
+::: tabs
+
+@tab Gradle
+
+Go to the root directory of the project from the terminal and run the below command
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:154]: Maybe add ":" at the end.

File: src/guide/frontend/android.md [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+
+@tab Powershell
+
+```powershell
+keytool -list -v -keystore "%USERPROFILE%\.android\debug.keystore" -alias androiddebugkey -storepass android -keypass android
+```
+
+:::
+
+Put this SHA256 along with your root android package name in your backend to generate `assetlinks.json` for your app at `https://yourexample.com/.well-known/assetlinks.json`.
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:174]: Should "android"  be capitalized here?

File: src/guide/frontend/android.md [hunk 15]
```diff
@@ -0,0 +1,340 @@
...
+If you are using `example-nodejs-backend`. then just put these 2 values in your `.env` file.
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:175]: I think that "." should be a ",".

File: src/guide/frontend/android.md [hunk 16]
```diff
@@ -0,0 +1,340 @@
...
+        )
+
+        return PasswordlessClient(options, activity, scope)
+    }
+}
+```
+
+@tab Without Dagger Hilt
+
+Or you can set the Context of PasswordlessClient manually: Ensure the context is set to the current `Activity`.
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:244]: I think that ":" should be a "." or a ",".

File: src/guide/frontend/android.md [hunk 17]
```diff
@@ -0,0 +1,340 @@
...
+_passwordless.setCoroutineScope(lifecycleScope)
+```
+
+:::
+
+## Registration
+
+1. **UI**:
+   - (Optional) Take the alias as input from the user if it's a non-discoverable credential.
+   - (Required) Specify the username for the credential.
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:266]: Would always recommend putting required steps above optional ones, unless the order they're done in matters.

File: src/guide/frontend/android.md [hunk 18]
```diff
@@ -0,0 +1,340 @@
...
+2. **Call `POST /register/token` implemented in `your backend`**: Your backend should call the Passwordless.dev API to generate a registration token.
+   - See (API documentation)[https://docs.passwordless.dev/guide/api.html#register-token]
+3. **Call `PasswordlessClient.register()`**
```
Comment (fschillingeriv) [inline src/guide/frontend/android.md:269]: Add ":" to match steps 1 & 2.

File: src/guide/frontend/android.md [hunk 19]
```diff
@@ -0,0 +1,340 @@
...
+   val backendUrl:String,
+
+   // Passwordless.dev server, change for self-hosting
+   val apiUrl: String = "https://v4.passwordless.dev"
+)
+```
+
+### .well-known/assetlinks.json
+
+In your application's `AndroidManifest.xml`, you will then need to add the tag below under `manifest::application`:
```
Comment (mmccabe4) [inline src/guide/frontend/android.md:74]: Little confusing to read, perhaps "add the follow tag under manifest::application"

File: src/guide/frontend/android.md [hunk 20]
```diff
@@ -0,0 +1,340 @@
...
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="assetlinks">https://yourexample.com/.well-known/assetlinks.json</string>
+</resources>
+```
+
+### Facet ID
+
+The `Facet ID` will be used at a later point in this guide to use as the `origin`.
```
Comment (mmccabe4) [inline src/guide/frontend/android.md:93]: This isnt an error necessarily, but sometimes shortening things can help them be more understandable. For example changing the line here to "will be used later in this guide as the `origin`. "

**Review by mmccabe4 — COMMENTED** [review]
just a few comments

File: src/guide/frontend/android.md [hunk 21]
```diff
@@ -0,0 +1,342 @@
...
+
+```kotlin
+data class PasswordlessOptions(
+   // Your public API key
+   val apiKey: String,
+
+   // Identifier for your server, for example 'example.com' if your backend is hosted at https://example.com.
+   val rpId: String,
+
+   // This is where your Facet ID goes
```
Comment (Tyrrrz) [inline src/guide/frontend/android.md:63]: Would be nice to have an example, similar to above, so the user knows what the value should look like

**Review by Tyrrrz — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Pull request merged**